### ISSUE TITLE
Logical operators whose result is a return, next or break statement produce invalid JavaScript

### ISSUE THREAD

**Issue by ticky** [opal/opal#2103]
Here’s a minimum viable reproduction of this case:

```ruby
true && return
```

It produces this JS code:
```ruby
/* Generated by Opal 1.0.3 */
(function(Opal) {
  var $a, self = Opal.top, $nesting = [], nil = Opal.nil,
  $$$ = Opal.const_get_qualified, $$ = Opal.const_get_relative,
  $breaker = Opal.breaker, $slice = Opal.slice, $truthy = Opal.truthy;

  return ($truthy($a = true) ? return nil; : $a)
})(Opal);
```

which, when parsed, emits this error:

```
Uncaught SyntaxError: Unexpected token ‘return'
```

Because JS `return` statements cannot appear within ternary expressions.

The same is true of Ruby `next` and `break` keywords in these situations, which also emit JS `return`s.

This was discovered while attempting to compile code which relies on [this part of Fugit::Cron](https://github.com/floraison/fugit/blob/7f4d5296e3e55e6c9853f14457af79d4b3e3a845/lib/fugit/cron.rb#L239-L261).

**Comment by elia** [comment]
Thanks for the simplified repro, that's always very appreciated!

This is more or less the same problem tracked in #810 and is tied to anything that breaks the control flow and is also syntactically an expression. In some cases that is solved with a throw (see e.g. `Opal.ret()` in `runtime.js`) and we'll probably need something like that here as well.

**Comment by hmdne** [comment]
Fixed by #2160

**Comment by ticky** [comment]
Nice, thank you! :)
